precompiles/types/coin.go

```diff
@@ -1,25 +1,25 @@
-package bank
+package types
 
 import (
 	"math/big"
 
 	"cosmossdk.io/math"
 	sdk "github.com/cosmos/cosmos-sdk/types"
 	"github.com/ethereum/go-ethereum/common"
-
-	ptypes "github.com/zeta-chain/node/precompiles/types"
 )
 
+const ZRC20DenomPrefix = "zrc20/"
+
 // ZRC20ToCosmosDenom returns the cosmos coin address for a given ZRC20 address.
 // This is converted to "zevm/{ZRC20Address}".
 func ZRC20ToCosmosDenom(ZRC20Address common.Address) string {
 	return ZRC20DenomPrefix + ZRC20Address.String()
 }
 
-func createCoinSet(tokenDenom string, amount *big.Int) (sdk.Coins, error) {
+func CreateCoinSet(tokenDenom string, amount *big.Int) (sdk.Coins, error) {
 	coin := sdk.NewCoin(tokenDenom, math.NewIntFromBigInt(amount))
 	if !coin.IsValid() {
-		return nil, &ptypes.ErrInvalidCoin{
+		return nil, &ErrInvalidCoin{
```

Comment:
Can the NewCoin function panic ? If so, should we consider handling it ?

Reply:
I don't think it can panic, the main problem with NewCoin is that it can return a nil non-valid coin, which is handled with `coin.IsValid()`.
I'll check further if it can raise a panic at some point, but I don't think so.

Reply:
After checking the NewCoin is panicable, added a guard.

Reply:
Applied in 06d0c5d5a8a2d7ce54f84a666c7bef7784963ccf